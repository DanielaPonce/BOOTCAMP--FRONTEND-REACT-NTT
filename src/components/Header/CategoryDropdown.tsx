import { FC } from 'react';
import { Category } from '../../models/category.types';

type Props = {
	categories: Category[];
	onSelectCategory: (category: string) => void;
};

const CategoryDropdown: FC<Props> = ({ categories, onSelectCategory }) => {
	return (
		<select
			onChange={(e) => onSelectCategory(e.target.value)}
			defaultValue="all"
		>
			<option value="all">Todas las categorias</option>

			{categories.map(({ name, slug }) => (
				<option key={slug} value={slug}>
					{name}
				</option>
			))}
		</select>
	);
};

export default CategoryDropdown;
