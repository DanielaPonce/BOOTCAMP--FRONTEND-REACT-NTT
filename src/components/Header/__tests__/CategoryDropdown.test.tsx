import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import CategoryDropdown from '../CategoryDropdown';
import { Category } from '../../../models/category.types';

describe('CategoryDropdown Component', () => {
	const mockCategories: Category[] = [
		{ name: 'Electronics', slug: 'electronics' },
		{ name: 'Books', slug: 'books' },
		{ name: 'Clothing', slug: 'clothing' }
	];

	const mockOnSelectCategory = jest.fn();

	beforeEach(() => {
		jest.clearAllMocks();
	});

	test('renders correctly', () => {
		render(
			<CategoryDropdown
				categories={mockCategories}
				onSelectCategory={mockOnSelectCategory}
			/>
		);

		const dropdown = screen.getByRole('combobox');
		expect(dropdown).toBeInTheDocument();

		const defaultOption = screen.getByText('Todas las categorias');
		expect(defaultOption).toBeInTheDocument();

		mockCategories.forEach(({ name }) => {
			expect(screen.getByText(name)).toBeInTheDocument();
		});
	});

	test('calls onSelectCategory with the correct value when a category is selected', () => {
		render(
			<CategoryDropdown
				categories={mockCategories}
				onSelectCategory={mockOnSelectCategory}
			/>
		);

		const dropdown = screen.getByRole('combobox');

		// Simula seleccionar la categorÃ­a "Books"
		fireEvent.change(dropdown, { target: { value: 'books' } });

		expect(mockOnSelectCategory).toHaveBeenCalledTimes(1);
		expect(mockOnSelectCategory).toHaveBeenCalledWith('books');
	});

	test('matches snapshot', () => {
		const { asFragment } = render(
			<CategoryDropdown
				categories={mockCategories}
				onSelectCategory={mockOnSelectCategory}
			/>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
