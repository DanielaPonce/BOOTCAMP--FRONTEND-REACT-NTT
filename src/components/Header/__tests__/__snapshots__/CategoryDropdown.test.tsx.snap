// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryDropdown Component matches snapshot 1`] = `
<DocumentFragment>
  <select>
    <option
      selected=""
      value="all"
    >
      Todas las categorias
    </option>
    <option
      value="electronics"
    >
      Electronics
    </option>
    <option
      value="books"
    >
      Books
    </option>
    <option
      value="clothing"
    >
      Clothing
    </option>
  </select>
</DocumentFragment>
`;
